### PR TITLE
[9.x] fixed issue on which class to check increment and decrement methods for custom cast

### PR DESCRIPTION
**Pull request by WendellAdriel**
This fix is related to the issue mentioned on #45418 
The code was checking if the increment/decrement methods were available on one class but then using another class to actually call the methods. This fix is for the code to check the same class that will be called the methods.

### CODE REVIEW

**Comment by Sjord** [comment]
Wouldn't it be better to just call the existing [`resolveCasterClass`](https://github.com/laravel/framework/blob/3dff5847ef8b9033ab1a414829abb543f1a9038a/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L1641-L1663) method?

**Comment by WendellAdriel** [comment]
I see your point @Sjord 

But the `resolveCasterClass` will resolve the class while in the code I wrote it's going to only check if the `castUsing` is present and get the class from it without creating an instance of the class.

I'm fine with both solutions tho. I'll just wait for someone to review it and if needed I can update it.
Thanks for the help 😉

**Comment by taylorotwell** [comment]
I think a test would be good here.

File: src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php [hunk 1]
```diff
@@ -1612,9 +1612,16 @@ protected function isEnumCastable($key)
...
-        return $this->isClassCastable($key) &&
-            method_exists($castType = $this->parseCasterClass($this->getCasts()[$key]), 'increment') &&
-            method_exists($castType, 'decrement');
+        if (! $this->isClassCastable($key)) {
+            return false;
+        }
+
+        $castType = $this->parseCasterClass($this->getCasts()[$key]);
+        if (method_exists($castType, 'castUsing')) {
+            $castType = $castType::castUsing([]);
```
Comment (sumihiran) [inline src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php:1621]: Wouldn't calling the "castUsing" method with an empty array disregards any given cast parameters?

**Comment by WendellAdriel** [comment]
@taylorotwell I'll work on the test
@sumihiran I'm going to check your observation

**Comment by WendellAdriel** [comment]
@taylorotwell I made some changes:
- Added a test case following the example given on #45418 
- Changed the `isClassDeviable` to use the `resolveCasterClass` method because of the observation made by @sumihiran

**Pull request merged**